ens/src/resolver.ts

```diff
@@ -0,0 +1,44 @@
+import * as airstack from "../modules/airstack/domain-name";
+import {
+  AddrChanged as AddrChangedEvent,
+  VersionChanged as VersionChangedEvent,
+} from "../generated/Resolver1/Resolver";
+import { log } from "@graphprotocol/graph-ts";
+import { ETHEREUM_MAINNET_ID } from "../modules/airstack/domain-name/utils";
+import { TOKEN_ADDRESS_ENS } from "./utils";
+/**
+ * @dev this function maps the AddrChanged event from the Resolver contract
+ * @param event AddrChangedEvent from Resolver contract
+ */
+export function handleAddrChanged(event: AddrChangedEvent): void {
+  log.info("handleAddrChanged: node {} addr {} resolver {} txhash {}", [event.params.node.toHexString(), event.params.a.toHexString(), event.address.toHexString(), event.transaction.hash.toHexString()]);
+  airstack.domain.trackAddrChangedTransaction(
+    ETHEREUM_MAINNET_ID,
+    event.logIndex,
+    event.address.toHexString(),
+    event.params.a.toHexString(),
+    event.params.node.toHexString(),
+    event.block.number,
+    event.block.hash.toHexString(),
+    event.block.timestamp,
+    event.transaction.hash.toHexString(),
+    TOKEN_ADDRESS_ENS,
+  );
+}
+
+/**
+ * @dev this function maps the VersionChanged event from the Resolver contract
+ * @param event VersionChangedEvent from Resolver contract
+ */
+export function handleVersionChanged(event: VersionChangedEvent): void {
+  log.info("handleVersionChanged: node {} resolver {} txhash {}", [event.params.node.toHexString(), event.address.toHexString(), event.transaction.hash.toHexString()]);
+  airstack.domain.trackResolverVersionChange(
```

Comment:
I think this can happen with any domain subgraph, the handler basically tracks the event when an owner changes the underlying resolver for a domain name. That sets the default resolved address for a domain as null.

Reply:
Lets confirm on this.

Reply:
I did some digging into unstoppable domains and was unable to find out any version changed events as per docs/ resolver/ registry contracts.

Reply:
Let's do this at the time of module refactoring.